Bail only if transaction does not make it on-chain

luckysori opened the pull request:
We no longer wrongly report an error when failing to broadcast a transaction because it's already on-chain.

Additionally, I tried to simplify this function with https://github.com/comit-network/hermes/pull/395/commits/3e8e6bc06eb99e504504c0a14c5f01ed1422a48d, which seems a lot better to me, but I ~could have~ **missed something**.

klochowicz commented on daemon/src/wallet.rs:
```diff
@@ -96,31 +96,33 @@ impl Wallet {
+                })?;
+
+                if error_code == i64::from(RpcErrorCode::RpcVerifyAlreadyInChain) {
+                    Ok(tx.txid())
```
perhaps add a log message stating that it's already on chain?

klochowicz reviewed (approved):
nice one 💯

luckysori committed:
Convert tracing::error into anyhow::Error context

This prevents us from reporting an error when the transaction was
actually on-chain.
